lib/tasks/manageiq/ui_tasks.rake

```diff
@@ -41,9 +26,15 @@ namespace :webpack do
 
   [:compile, :clobber].each do |webpacker_task|
     task webpacker_task do
-      Dir.chdir ManageIQ::UI::Classic::Engine.root do
-        Rake::Task["webpack:paths"].invoke
-        Rake::Task["webpacker:#{webpacker_task}"].invoke
+      # Run the `webpack:compile` tasks without a fully loaded environment,
+      # since when doing an appliance/docker build, a database isn't
+      # available for the :environment task (prerequisite for
+      # 'webpacker:compile') to function.
+      EvmRakeHelper.with_dummy_database_url_configuration do
+        Dir.chdir ManageIQ::UI::Classic::Engine.root do
+          Rake::Task["webpack:paths"].invoke
+          Rake::Task["webpacker:#{webpacker_task}"].invoke
+        end
```

Comment:
So I think my rational for the `defined?` is that if you run the `rake webpack:compile` from this repo (not `manageiq` proper, and also not the `rake app:webpack:compile` task), then the task wouldn't work, since it wouldn't load the codebase from `manageiq`.  That said, I haven't tried it recently to confirm if it is a problem or not.

If it is, to avoid duplication, you could possibly do this:

```ruby
task_proc = lambda do
  Dir.chdir ManageIQ::UI::Classic::Engine.root do
    Rake::Task["webpack:paths"].invoke
    Rake::Task["webpacker:#{webpacker_task}"].invoke
  end
end

if defined?(EvmRakeHelper)
  EvmRakeHelper.with_dummy_database_url_configuration { task_proc.call }
else
  task_proc.call
end
```

But again, that is only if my theory about the `rake webpack:compile` from this repo is true.

Reply:
Great idea with the proc, thanks :). (I was actually thinking gaah, I wish I could add an anonymous function... didn't think of a proc.)

That said, I can verify that running `rake webpack:compile` from `manageiq-ui-classic` does load `manageiq/lib/tasks/evm_rake_helper.rb`.

So.. I guess we don't need it?

Reply:
Meh, I would then say you are fine (pun not intended).  

That said, does this need to be backported at all?  If so, I am hesitant to remove the `defined?` so close to the build date.  But if we are just pushing this to master, then I think it is okay to keep what you have here, and let things break out in the wild if they do.

Reply:
:+1: added `gaprindashvili/no`.. nothing is actually biting us here, I just had a note in my TODO :).